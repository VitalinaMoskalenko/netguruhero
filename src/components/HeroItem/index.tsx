import React from "react";
import styled from "styled-components";
import { placeholderAvatar } from "../../assets";
import { H3 } from "../Headings";
import { Body } from "../Paragraphs";

const Container = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background-color: ${({ theme }) => theme.colors.white};
  height: 50px;
  padding: 18px;
  border-radius: 8px;
`;

const HeroContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

const Avatar = styled.img`
  width: 40px;
  height: 40px;
  border-radius: 20px;
`;

const Name = styled(H3)`
  margin-left: 20px;
`;

type PropsType = {
  imageUrl?: string;
  name: string;
  type: string;
  description: string;
};

const HeroItem = ({ imageUrl, name, type, description }: PropsType) => {
  return (
    <Container>
      <HeroContainer>
        <Avatar src={imageUrl || placeholderAvatar} />
        <Name>{name}</Name>
      </HeroContainer>
      <Body>{type}</Body>
      <Body>{description}</Body>
    </Container>
  );
};

export default HeroItem;
