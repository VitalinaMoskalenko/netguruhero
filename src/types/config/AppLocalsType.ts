export enum AppLocalsType {
  en = "en",
}
