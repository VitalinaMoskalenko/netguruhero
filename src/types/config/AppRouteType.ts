export enum AppRouteType {
  home = "/",
  addHeroForm = "/addHeroForm",
  heroDetails = "/heroDetails",
  heroDetailsWithId = "/heroDetails/:id",
}
