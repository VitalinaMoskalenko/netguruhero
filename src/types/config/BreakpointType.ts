export enum BreakpointType {
  xxs = 375,
  xs = 768,
  xxl = 1440,
}
