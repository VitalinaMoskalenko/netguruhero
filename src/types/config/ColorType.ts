export enum ColorType {
  white = "white",
  black = "black",
  green = "#63cd93",
  blue = "#6199f3",
  gray = "#f5f6fb",
  black05 = "rgba(0,0,0, 0.5)",
}
