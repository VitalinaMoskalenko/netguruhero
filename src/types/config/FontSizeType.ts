export enum FontSizeType {
  smallBody = 14,
  body = 18,
  h5 = 16,
  h4 = 18,
  h3 = 24,
  h2 = 32,
  h1 = 48,
}
