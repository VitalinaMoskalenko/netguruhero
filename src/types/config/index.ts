export * from "./AppRouteType";
export * from "./FontSizeType";
export * from "./ColorType";
export * from "./BreakpointType";
export * from "./AppLocalsType";
