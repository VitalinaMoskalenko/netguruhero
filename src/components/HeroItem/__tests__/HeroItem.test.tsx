import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import HeroItem from "../";
import { theme } from "../../../config";

it("renders hero item correctly", () => {
  const heroItemComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <HeroItem imageUrl="" name="" description="" type="" />
      </ThemeProvider>
    )
    .toJSON();

  expect(heroItemComponent).toMatchSnapshot();
});
