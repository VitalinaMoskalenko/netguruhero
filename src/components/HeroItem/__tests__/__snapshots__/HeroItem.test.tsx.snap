// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders hero item correctly 1`] = `
<div
  className="sc-iCoGMd jTODGz"
>
  <div
    className="sc-fujyAs hMTAzb"
  >
    <img
      className="sc-pNWdM jxqPfb"
      src="placeholderAvatar.png"
    />
    <div
      className="sc-iqAclL bsILWn"
    >
      <h1
        className="sc-dlnjwi czuNcg"
      >
        
      </h1>
      <p
        className="sc-jSFjdj sc-kEqXSa bQuzKy btUuiI"
      >
        
      </p>
    </div>
  </div>
  <p
    className="sc-jSFjdj sc-jrsJWt bQuzKy hOSzWK"
  >
    
  </p>
  <p
    className="sc-jSFjdj sc-crzoAE bQuzKy fqRwMk"
  >
    
  </p>
</div>
`;
